File: nodeApp-Vue/routes/httpTest.js
const express = require('express');
const router = express.Router();
const db = require("../my_modules/db.js");
const moment = require('moment');

// http://localhost:3000/nodejs/httpTest?action=findData&searchJson={"name":"mick"}&fieldJson={"name":1}&prePageNum=10&currPage=1&sortJson={"key":1}  //查询数据
// http://localhost:3000/nodejs/httpTest?action=insertData&dataArr=[{"name":"mick","age":18},{"name":"tina","age":35}]  //插入数据 
// http://localhost:3000/nodejs/httpTest?action=updateData&searchJson={"name":"mick"}&updateJson={"age":20}  //修改数据 
// http://localhost:3000/nodejs/httpTest?action=delData&dataJson={"name":{"$in":["mick","tina"]}}  //删除数据 
// http://localhost:3000/nodejs/httpTest?action=addCol&dataJson={"name":""}  //插入列
// http://localhost:3000/nodejs/httpTest?action=updateCol&dataJson={"oldname":"newname"}  //修改列
// http://localhost:3000/nodejs/httpTest?action=delCol&dataJson={"name":""}  //删除列



//创建索引
//db.createIndex(dbName,{"id":1},function(err,result){
//      if(err){
//          console.log('createIndex:'+ err);
//      }
//      //console.log(result);
//});


//创建数据表
//db.createCollection(dbName,function(err,result){
//      if(err){
//          console.log('createCollection:'+ err);
//      }
//      //console.log(result);
//});


//删除数据表
//db.dropCollection(dbName,function(err,result){
//      if(err){
//          console.log('dropCollection:'+ err);
//      }
//      //console.log(result);
//});




router.get('/', function(req, res, next) {
	
	var dbName = req.baseUrl.split("/")[2];
	
	//返回错误
    function returnErr (value) {
    	console.log(value);
		res.send(value);
    }
    
    
    //分配方法
	console.log(req.query);
	var action = req.query.action;
	switch (action){
		case "findData":
		      findData();
			  break;
        default:
              returnErr("action错误");
	}
	
    


	//查询数据 
	function findData () {
		if(req.query.searchJson){
			try{
			  var whereJson = JSON.parse(req.query.searchJson);
			}catch(e){
				returnErr("searchJson错误");
				return;
			}
		}else{
			var whereJson = {};
		}
		
		if(req.query.sortJson){
			try{
			  var sortJson = JSON.parse(req.query.sortJson);
			}catch(e){
				returnErr("sortJson错误");
				return;
			}
		}else{
			var sortJson = {};
		}
		
		if(req.query.fieldJson){
			try{
			  var fieldJson = JSON.parse(req.query.fieldJson);
			}catch(e){
				returnErr("fieldJson错误");
				return;
			}
		}else{
			var fieldJson = {};
		}
		
		var prePageNum = req.query.prePageNum ? Number(req.query.prePageNum) : 0;
		var currPage = req.query.currPage ? Number(req.query.currPage) : 0;
		var args = {"prePageNum":prePageNum, "currPage":currPage, "sort":sortJson};
		db.findData(dbName, whereJson, fieldJson, args, function(err, result, count){
		        if(err){
		            returnErr('findDataError:'+ err);
				    return;
		        }
		       //console.log(result);
		       console.log("查询了" + result.length +"条数据，" + "共" + count +"条数据");
		       var jsonObj = {"code": 0, "msg" : "", "count" : count, "prePageNum": prePageNum, "currPage": currPage, "rows": result};
		       res.send(jsonObj);
		});
	}
	
	
	
});

	
	
router.post('/', function(req, res, next) {
	
	var dbName = req.baseUrl.split("/")[2];
	
	
	//返回错误
    function returnErr (value) {
    	console.log(value);
		res.send(value);
    }
    
    
    //分配方法
	console.log(req.body);
	var action = req.body.action;
	switch (action){
	    case "insertData":
		      insertData();
			  break;
		case "updateData":
		      updateData();
			  break;
		case "delData":
		      delData();
			  break;
		case "addCol":
		      addCol();
			  break;
		case "updateCol":
		      updateCol();
			  break;
		case "delCol":
		      delCol();
			  break;
        default:
              returnErr("action错误");
              
           
	}
	
    
    

	
	//插入数据 
	function insertData () {
		if(req.body.dataArr){
			try{
			    var dataArr = JSON.parse(req.body.dataArr);
			    if(Array.isArray(dataArr) && dataArr.length > 0){
					//continue
				}else{
					returnErr("dataArr错误");
					return;
				}
			}catch(e){
				returnErr("dataArr错误");
				return;
			}
		}else{
			returnErr("dataArr错误");
			return;
		}
		var idArr = [];
		for(let i=0; i<dataArr.length; i++){
			dataArr[i].id = dataArr[i].id;
			dataArr[i].create_name = req.cookies["logining"].username;
            dataArr[i].create_time = moment().format("YYYY-MM-DD HH:mm:ss");
            idArr.push(dataArr[i].id);
		}
		var whereJson = {"id":{"$in":idArr}};
		var fieldJson = {"id": 1};
		var args = "";
		db.findData(dbName, whereJson, fieldJson, args, function(err, result, count){
		        if(err){
		            returnErr('findDataError:'+ err);
				    return;
		        }
		       //console.log(result);
		       if(result.length > 0){
		       	 res.send("ID已存在");
				 return;
		       }
		       db.insertData(dbName,dataArr,function(err2,result2){
			        if(err2){
			            returnErr('insertDataError:'+ err2);
					    return;
			        }
			        console.log("插入了" + result2.result.n +"条数据");
			        if(result2.result.n > 0){
			        	res.send("操作成功");
			        }else{
			        	res.send("操作失败");
			        }
			        
			   });
		});

	}
	
	
	//修改数据 
	function updateData () {
		if(req.body.searchJson){
			try{
			  var whereJson = JSON.parse(req.body.searchJson);
			}catch(e){
				returnErr("searchJson错误");
				return;
			}
		}else{
			returnErr("searchJson错误");
			return;
		}
		
		if(req.body.updateJson){
			try{
			  var updateData = JSON.parse(req.body.updateJson);
			}catch(e){
				returnErr("updateJson错误");
				return;
			}
		}else{
			returnErr("updateJson错误");
			return;
		}
		
		updateData.update_name = req.cookies["logining"].username;
        updateData.update_time = moment().format("YYYY-MM-DD HH:mm:ss");
		var updateJson = {$set: updateData};
		db.updateData(dbName, whereJson, updateJson, function(err,result){
		        if(err){
		            returnErr('updateDataError:'+ err);
				    return;
		        }
		        console.log("修改了" + result.result.n +"条数据");
		        if(result.result.n > 0){
		        	res.send("操作成功");
		        }else{
		        	res.send("操作失败");
		        }
		});
	}
  

    //删除数据 
	function delData () {
		if(req.body.dataJson){
			try{
			  var whereJson = JSON.parse(req.body.dataJson);
			}catch(e){
				returnErr("delJson错误");
				return;
			}
		}else{
			returnErr("delJson错误");
			return;
		}
		db.deleteData(dbName, whereJson, function(err,result){
		        if(err){
		            returnErr('deleteDataError:'+ err);
				    return;
		        }
		        console.log("删除了" + result.result.n +"条数据");
		        if(result.result.n > 0){
		        	res.send("操作成功");
		        }else{
		        	res.send("操作失败");
		        }
		});
	}
	
	
	//插入列
	function addCol () {
		if(req.body.dataJson){
			try{
			  var dataJson = JSON.parse(req.body.dataJson);
			}catch(e){
				returnErr("dataJson错误");
				return;
			}
		}else{
			returnErr("dataJson错误");
			return;
		}
		var whereJson = {};
		var updateJson = {$set: dataJson};
		db.updateCol(dbName, whereJson, updateJson, function(err,result){
		        if(err){
		            returnErr('updateCol:'+ err);
				    return;
		        }
		        var arr =[];
		        for(var key in dataJson){
		        	arr.push("插入列" + key );
		        }
		        console.log(arr.join(",") +"共" + result.result.n +"条数据");
		        if(result.result.n > 0){
		        	res.send("操作成功");
		        }else{
		        	res.send("操作失败");
		        }
		});
	}
	
	
	
	
	//修改列
	function updateCol () {
		if(req.body.dataJson){
			try{
			  var dataJson = JSON.parse(req.body.dataJson);
			}catch(e){
				returnErr("dataJson错误");
				return;
			}
		}else{
			returnErr("dataJson错误");
			return;
		}
		var whereJson = {};
		var updateJson = {$rename: dataJson};
		db.updateCol(dbName, whereJson, updateJson, function(err,result){
		        if(err){
		            returnErr('updateCol:'+ err);
				    return;
		        }
		        var arr =[];
		        for(var key in dataJson){
		        	arr.push("修改列名" + key + "为"+ dataJson[key]);
		        }
		        console.log(arr.join(",") +"共" + result.result.n +"条数据");
		        if(result.result.n > 0){
		        	res.send("操作成功");
		        }else{
		        	res.send("操作失败");
		        }
		});
	}
	
	
	
	//删除列
	function delCol () {
		if(req.body.dataJson){
			try{
			  var dataJson = JSON.parse(req.body.dataJson);
			}catch(e){
				returnErr("dataJson错误");
				return;
			}
		}else{
			returnErr("dataJson错误");
			return;
		}
		var whereJson = {};
		var updateJson = {$unset: dataJson};
			db.updateCol(dbName, whereJson, updateJson, function(err,result){
		        if(err){
		            returnErr('updateCol:'+ err);
				    return;
		        }
		        var arr =[];
		        for(var key in dataJson){
		        	arr.push("删除列" + key );
		        }
		        console.log(arr.join(",") +"共" + result.result.n +"条数据");
		        if(result.result.n > 0){
		        	res.send("操作成功");
		        }else{
		        	res.send("操作失败");
		        }
		});
	}
	
	
	
	
	
	
	
});

	



module.exports = router;

File: flaskApp-LayUI/flaskApp/views/http_test.py
# -*- coding: utf-8 -*-
from flask import make_response
from flaskApp.my_modules import mysqldb
import json
import time


#http://localhost:3000/python/http_test?action=findData&whereStr=id=1 and name="xx"&fieldStr=field1,field2&prePageNum=10&currPage=1&sortStr=id ASC|DESC  //查询数据
#http://localhost:3000/python/http_test?action=insertData&dataArr=[{"name":"mick","age":18},{"name":"tina","age":35}]  //插入数据
#http://localhost:3000/python/http_test?action=updateData&whereStr=id=1&updateJson={"name":"xx"}  //修改数据
#http://localhost:3000/python/http_test?action=delData&dataArr=[1,3,5]  //删除数据
#http://localhost:3000/python/http_test?action=addCol&dataJson={"fieldName":"num","dataType":"INT"}  //插入列
#http://localhost:3000/python/http_test?action=updateCol&dataJson={"fieldName":"num","fieldNewName":"meg","dataNewType":"TEXT"}  //修改列
#http://localhost:3000/python/http_test?action=delCol&dataJson={"fieldName":"num"}  //删除列


def operation(req):
    table_name = req.path[8:]
    dict_login = json.loads(req.cookies['logining'])

    # 查询数据
    def find_data():

        if 'whereStr' in req.args:
            str_where = req.args['whereStr']
        else:
            str_where = ''

        if 'fieldStr' in req.args:
            str_field = req.args['fieldStr']
        else:
            str_field = ''

        if 'sortStr' in req.args:
            str_sort = req.args['sortStr']
        else:
            str_sort = ''

        if 'prePageNum' in req.args:
            pre_page_num = int(req.args['prePageNum'])
        else:
            pre_page_num = 0

        if 'currPage' in req.args:
            curr_page = int(req.args['currPage'])
        else:
            curr_page = 0

        args = {'pre_page_num': pre_page_num, 'curr_page': curr_page, 'sort': str_sort}
        result = mysqldb.find_data(table_name, str_where, str_field, args)
        # print(result)

        if result:
            dict_json = {'code': 0, 'msg': '', 'count': result['count'], 'prePageNum': pre_page_num, 'currPage': curr_page, 'rows': result['rows']}
            return make_response(json.dumps(dict_json))
        else:
            return make_response('操作失败')


    # 插入数据
    def insert_data():
        if 'dataArr' in req.form:
            try:
                list_data = json.loads(req.form['dataArr'])
                if len(list_data) == 0:
                    return make_response('dataArr错误')
            except:
                return make_response('dataArr错误')
        else:
            return make_response('dataArr错误')

        for item in list_data:
            item['create_name'] = dict_login['username']
            item['create_time'] = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())

        result = mysqldb.insert_data(table_name, list_data)
        # print(result)

        if result:
            return make_response('操作成功')
        else:
            return make_response('操作失败')


    # 修改数据
    def update_data():
        if 'whereStr' in req.form:
            str_where = req.form['whereStr']
            if not str_where:
                return make_response('whereStr错误')
        else:
            return make_response('whereStr错误')

        if 'updateJson' in req.form:
            try:
                dict_update = json.loads(req.form['updateJson'])
                if len(dict_update) == 0:
                    return make_response('updateJson错误')
            except:
                return make_response('updateJson错误')
        else:
            return make_response('updateJson错误')

        dict_update['update_name'] = dict_login['username']
        dict_update['update_time'] = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())

        result = mysqldb.update_data(table_name, str_where, dict_update)
        # print(result)

        if result:
            return make_response('操作成功')
        else:
            return make_response('操作失败')


    # 删除数据
    def del_data():
        if 'dataArr' in req.form:
            try:
                list_data = json.loads(req.form['dataArr'])
                if len(list_data) == 0:
                    return make_response('dataArr错误')
            except:
                return make_response('dataArr错误')
        else:
            return make_response('dataArr错误')


        result = mysqldb.del_data(table_name, list_data)
        # print(result)

        if result:
            return make_response('操作成功')
        else:
            return make_response('操作失败')


    # 插入列
    def add_col():
        if 'dataJson' in req.form:
            try:
                dict_data = json.loads(req.form['dataJson'])
            except:
                return make_response('dataJson错误')
        else:
            return make_response('dataJson错误')

        if not dict_data['fieldName'] and not dict_data['dataType']:
            return make_response('dataJson错误')

        str_data = 'add ' + dict_data['fieldName'] + ' ' + dict_data['dataType'] + ' not null'
        result = mysqldb.update_col(table_name, str_data)
        # print(result)

        if result:
            return make_response('操作成功')
        else:
            return make_response('操作失败')

    # 修改列
    def update_col():
        if 'dataJson' in req.form:
            try:
                dict_data = json.loads(req.form['dataJson'])
            except:
                return make_response('dataJson错误')
        else:
            return make_response('dataJson错误')

        if not dict_data['fieldName'] and not dict_data['fieldNewName']:
            return make_response('dataJson错误')

        str_data = 'change ' + dict_data['fieldName'] + ' ' + dict_data['fieldNewName']
        if dict_data['dataNewType']:
            str_data += ' ' + dict_data['dataNewType']

        result = mysqldb.update_col(table_name, str_data)
        # print(result)

        if result:
            return make_response('操作成功')
        else:
            return make_response('操作失败')

    # 删除列
    def del_col():
        if 'dataJson' in req.form:
            try:
                dict_data = json.loads(req.form['dataJson'])
            except:
                return make_response('dataJson错误')
        else:
            return make_response('dataJson错误')

        if not dict_data['fieldName']:
            return make_response('dataJson错误')

        str_data = 'drop ' + dict_data['fieldName']

        result = mysqldb.update_col(table_name, str_data)
        # print(result)

        if result:
            return make_response('操作成功')
        else:
            return make_response('操作失败')


    # GET请求
    if req.method == 'GET':
        print(req.args)
        if req.args['action'] == 'findData':
            return find_data()
        else:
            return make_response('action错误')

    # POST请求
    if req.method == 'POST':
        print(req.form)
        if req.form['action'] == 'insertData':
            return insert_data()
        elif req.form['action'] == 'updateData':
            return update_data()
        elif req.form['action'] == 'delData':
            return del_data()
        elif req.form['action'] == 'addCol':
            return add_col()
        elif req.form['action'] == 'updateCol':
            return update_col()
        elif req.form['action'] == 'delCol':
            return del_col()
        else:
            return make_response('action错误')

    return make_response('action错误')